README.md

```diff
@@ -43,13 +43,63 @@ Finally, generate an invite link for the bot and enter it into any browser.
 > [!NOTE]
 > Bot permissions involving threads/mentions/messages should suffice, but you can naively give it `Administrator` since it's just a test bot in your own testing Discord server.  
 
+### Database
+
+The leaderboard persists information in a Postgres database. To develop locally, set Postgres up on your machine. Then create a database with:
+
+```
+CREATE DATABASE clusterdev;
+```
+
+We are using [Yoyo Migrations](https://ollycope.com/software/yoyo/) to manage tables, indexes, etc. in our database. To create tables in your local database, apply the migrations in `src/discord-cluster-manager/migrations` with the following command line:
+
+```
+yoyo apply src/discord-cluster-manager/migrations \
+  -d postgresql://user:password@localhost/clusterdev
+```
+
+  <details>
+    <summary>Click here for a transcript of a yoyo apply session</summary>
+
+    $ yoyo apply . -d postgresql://user:password@localhost/clusterdev
+
+    [20241208_01_p3yuR-initial-leaderboard-schema]
+    Shall I apply this migration? [Ynvdaqjk?]: y
+
+    Selected 1 migration:
+      [20241208_01_p3yuR-initial-leaderboard-schema]
+    Apply this migration to postgresql://user:password@localhost/clusterdev [Yn]: y
```

Comment:
sorry n00b q: wdym by migration here?

Reply:
So the term "migrations" refers to a type of version control for databases. The general idea is to specify changes to a database schema with a sequence of files. These files contain instructions, often in the form of SQL statements, that change database structure. As the application evolves, new migration files get added, so that the database structure changes along with the application. That that the files are also stored in version control provides a historical view of how the database changed in structure over time. 

A very typical use case is to add a column as part of a migration, and then to populate the column using existing data.

There is a whole family of tools for migrations in different environments. For example, Ruby on Rails has ActiveRecord migrations, Java has Liquibase and Flyway, Python has Alembic and Yoyo, and so on. The common flavor of these tools is that they allow CI/CD systems (or, in the old days, devops) to apply migrations in the same order across different environments (local dev, staging, prod, etc.).